### PR TITLE
chore: Enable strict mode in `aws-amplify`

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change enabled strict mode in `aws-amplify`.

#### Description of how you validated changes
Local testing.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify/src/initSingleton.ts [hunk 1]
```diff
@@ -14,30 +14,26 @@ import {
...
 export const DefaultAmplify = {
 	configure(resourceConfig: ResourcesConfig, libraryOptions?: LibraryOptions) {
-		CognitoUserPoolsTokenProvider.setAuthConfig(resourceConfig.Auth);
-		const defaultLibraryOptions: LibraryOptions = {
-			Auth: {
-				tokenProvider: CognitoUserPoolsTokenProvider,
-				credentialsProvider: cognitoCredentialsProvider,
-			},
-		};
+		if (resourceConfig.Auth && !libraryOptions) {
```
Comment (israx) [inline packages/aws-amplify/src/initSingleton.ts:17]: I see with this change the code bellow will only be executed if there are not `libraryOptions`, whereas the previous code will assign `libraryOptions` to the config if they are passed or not.

**Review by israx — COMMENTED** [review]
Looks pretty good. Just a question about whether the intention of this change is to only set `libraryOptions` within the library

File: packages/aws-amplify/src/initSingleton.ts (hunk 1 above)
Comment (jimblanc) [reply packages/aws-amplify/src/initSingleton.ts:17]: Good catch, added a fallback case

**Review by israx — APPROVED** [review]
lgtm!

**Review by stocaaro — APPROVED** [review]

**Pull request merged**